resources/xsd/fastRTPS_profiles.xsd

```diff
@@ -309,867 +1140,432 @@
     </xs:complexType>
 
-    <xs:simpleType name="ownershipQosKindType">
-        <xs:restriction base="xs:string">
-            <xs:enumeration value="SHARED"/>
-            <xs:enumeration value="EXCLUSIVE"/>
-        </xs:restriction>
-    </xs:simpleType>
-
+    <!--QoS Ownership:
+        └ kind [string] ("SHARED", "EXCLUSIVE") -->
     <xs:complexType name="ownershipQosPolicyType">
         <xs:all>
-            <xs:element name="kind" type="ownershipQosKindType"/>
+            <xs:element name="kind" minOccurs="0" maxOccurs="1">
+                <xs:simpleType>
+                    <xs:restriction base="xs:string">
+                        <xs:enumeration value="SHARED"/>
+                        <xs:enumeration value="EXCLUSIVE"/>
+                    </xs:restriction>
+                </xs:simpleType>
+            </xs:element>
         </xs:all>
     </xs:complexType>
 
+    <!--QoS Ownership Strength:
+        └ value [uint32] -->
     <xs:complexType name="ownershipStrengthQosPolicyType">
         <xs:all>
-            <xs:element name="value" type="uint32Type"/>
+            <xs:element name="value" type="uint32" minOccurs="0" maxOccurs="1"/>
         </xs:all>
     </xs:complexType>
 
-    <xs:simpleType name="destinationOrderQosKindType">
-        <xs:restriction base="xs:string">
-            <xs:enumeration value="BY_RECEPTION_TIMESTAMP"/>
-            <xs:enumeration value="BY_SOURCE_TIMESTAMP"/>
-        </xs:restriction>
-    </xs:simpleType>
-
-    <xs:complexType name="destinationOrderQosPolicyType">
-        <xs:all>
-            <xs:element name="kind" type="destinationOrderQosKindType"/>
-        </xs:all>
-    </xs:complexType>
-
-    <xs:simpleType name="presentationQosKindType">
-        <xs:restriction base="xs:string">
-            <xs:enumeration value="INSTANCE"/>
-            <xs:enumeration value="TOPIC"/>
-            <xs:enumeration value="GROUP"/>
-        </xs:restriction>
-    </xs:simpleType>
-
-    <xs:complexType name="presentationQosPolicyType">
-        <xs:all minOccurs="0">
-            <xs:element name="access_scope" type="presentationQosKindType" minOccurs="0"/>
-            <xs:element name="coherent_access" type="boolType" minOccurs="0"/>
-            <xs:element name="ordered_access" type="boolType" minOccurs="0"/>
-        </xs:all>
-    </xs:complexType>
-
-    <xs:complexType name="nameVectorType">
-        <xs:sequence>
-            <xs:element name="name" type="stringType" maxOccurs="unbounded"/>
-        </xs:sequence>
-    </xs:complexType>
-
+    <!--QoS Partition:
+        └ names    [1~*],
+            └ name [string] -->
     <xs:complexType name="partitionQosPolicyType">
         <xs:all>
-            <xs:element name="names" type="nameVectorType"/>
+            <xs:element name="names" minOccurs="1" maxOccurs="1">
+                <xs:complexType>
+                    <xs:sequence>
+                        <xs:element name="name" type="string" minOccurs="1" maxOccurs="unbounded"/>
```

Comment:
I have tried erasing the value inside a "name" tag, and the validator says it is valid.